flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -196,9 +207,14 @@ public void checkpointState(
 		// We generally try to emit the checkpoint barrier as soon as possible to not affect downstream
 		// checkpoint alignments
 
+		if (lastCheckpointId >= metadata.getCheckpointId()) {
+			LOG.warn("Out of order checkpoint barrier (aborted previously?): {} >= {}", lastCheckpointId, metadata.getCheckpointId());
```

Comment:
Should that be warning? I think on concept level yes.
However, in reality, in current UC, this case happens probably quite frequently (barrier overtakes by cancellation marker doesn't). I wouldn't like seeing the log polluted with WARN.